rllib/examples/ray_tune/custom_progress_reporter.py

```diff
@@ -0,0 +1,116 @@
+how the (frequent) results are being formatted and written to e.g. a logfile.
+To see an example on how to write your own Logger, see:
+https://github.com/ray-project/ray/tree/master/rllib/examples/ray_tune/custom_logger.py
+
+
+How to run this script
+----------------------
+`python [script file name].py
+
+
+Results to expect
+-----------------
+You should see something similar to the following in your console output:
+
++---------------------+------------+-----------------+--------+------------------+
+| Trial name          | status     | loc             |   iter |   total time (s) |
+|---------------------+------------+-----------------+--------+------------------+
+| PPO_env_bb503_00000 | TERMINATED | 127.0.0.1:26303 |      5 |          30.3823 |
++---------------------+------------+-----------------+--------+------------------+
++-------+-------------------+------------------+------------------+------------------+
+|    ts |   combined return |   return policy1 |   return policy2 |   return policy3 |
+|-------+-------------------+------------------+------------------+------------------|
+| 20000 |             258.7 |            103.4 |            88.84 |            87.86 |
++-------+-------------------+------------------+------------------+------------------+
+
+"""
+from ray import air, tune
+from ray.rllib.algorithms.ppo import PPOConfig
+from ray.rllib.env.multi_agent_env_runner import MultiAgentEnvRunner
+from ray.rllib.examples.envs.classes.multi_agent import MultiAgentCartPole
+
+
+my_multi_agent_progress_reporter = tune.CLIReporter(
+    # In the following dict, the keys are the (possibly nested) keys that can be found
+    # in RLlib's (PPO's) result dict, produced at every training iteration, and the
+    # values are the column names you would like to see in your console reports.
+    # Note that for nested result dict keys, you need to use slashes "/" to define the
+    # exact path.
+    metric_columns={
+        **{
+            "training_iteration": "iter",
+            "time_total_s": "total time (s)",
+            "timesteps_total": "ts",
+            # RLlib always sums up all agents' rewards and reports it under:
+            # result_dict[sampler_results][episode_reward_mean].
+            "sampler_results/episode_reward_mean": "combined return",
+        },
+        # Because RLlib sums up all returns of all agents, we would like to also
+        # see the individual agents' returns. We can find these under the result dict's
+        # 'policy_reward_mean' key (then the policy ID):
+        **{
+            f"policy_reward_mean/{pid}": f"return {pid}"
+            for pid in ["policy1", "policy2", "policy3"]
+        },
+    },
+)
+
+
+if __name__ == "__main__":
+    # Force Tuner to use old progress output as the new one silently ignores our custom
+    # `CLIReporter`.
+    # TODO (sven): Find out why we require this hack.
+    import os
+
+    os.environ["RAY_AIR_NEW_OUTPUT"] = "0"
```

Comment:
I guess it would help users, if we describe in one line why we set this env variable.

Reply:
We do this in the line above:
```
# Force Tuner to use old progress output as the new one silently ignores our custom
# `CLIReporter`.
```